Objective-C/CBLDatabase.mm

```diff
@@ -681,12 +681,12 @@ - (nullable CBLScope*) scopeWithName: (NSString*)name error: (NSError**)error {
 
 - (nullable CBLCollection*) defaultCollection: (NSError**)error {
     CBL_LOCK(_mutex) {
-        // db closed or deleted
+        // db closed
         if (![self mustBeOpen: error])
             return nil;
         
-        // collection removed
-        BOOL isValid = _defaultCollection.isValid;
+        // if collection is invalid (deleted or LC marked as invalid)
+        BOOL isValid = [_defaultCollection collectionIsValid: error];
```

Comment:
It doesn't seem like passing error here is having any effects as it will get converted to NotFound anyway. So if it doesn't do anything at the end, just pass nil.